test/integration.spec.ts

```diff
@@ -206,6 +254,103 @@ describe("Integration Tests", () => {
     assert.deepEqual(symbolTable.get("n"), new NumberType());
   });
 
+  it("Lists: nested heterogenous lists + instance methods (good and bad cases)", () => {
+    let symbolTable = typecheckFiles([
+      "./test/test-examples/lists-nested-methods.js",
+    ]).getMap();
+
+    // console.log(report.getErrors());
+    // console.log(symbolTable);
+    assert.equal(report.getErrors().length, 1, "Expected 1 error generated");
```

Comment:
In tests where we are expecting an error, can we check if the error is the one we are expecting?

Reply:
I can do that, though it'll fix the error text to what we have so far.

Reply:
I think it would be better to have failing tests when someone changes an error message than not knowing if we are throwing the right error